pkg/proxy/service/cluster_service.go

```diff
@@ -41,6 +45,44 @@ func (s *ServiceImpl) GetClusterAccess(userID, username string) (*access.Cluster
 		return nil, errs.New("user is not provisioned (yet)")
 	}
 
+	// if workspace is not provided then return the default space access
+	if workspace == "" {
+		return s.accessForCluster(signup.APIEndpoint, signup.ClusterName, signup.CompliantUsername)
+	}
+
+	// look up space
+	space, err := s.Services().InformerService().GetSpace(workspace)
+	if err != nil {
+		// log the actual error but do not return it so that it doesn't reveal information about a space that may not belong to the requestor
+		log.Error(ctx, err, "unable to get target cluster")
+		return nil, fmt.Errorf("the requested space in not available")
+	}
+
+	return s.accessForSpace(space.Status.TargetCluster, signup.CompliantUsername)
+}
+
+func (s *ServiceImpl) accessForSpace(targetCluster, username string) (*access.ClusterAccess, error) {
+	// Get the target member
+	members := s.GetMembersFunc()
+	if len(members) == 0 {
+		return nil, errs.New("no member clusters found")
+	}
+	for _, member := range members {
+		if member.Name == targetCluster {
+			apiURL, err := url.Parse(member.APIEndpoint)
+			if err != nil {
+				return nil, err
+			}
+			// requests use impersonation so are made with member ToolchainCluster token, not user tokens
+			impersonatorToken := member.RestConfig.BearerToken
+			return access.NewClusterAccess(*apiURL, impersonatorToken, username), nil
+		}
+	}
+
+	return nil, errs.New("no member cluster found for the space")
```

Comment:
Let's maybe also log the targetCluster in case it's not found?
It's for easier debugging in case some space doesn't have Status.TargetCluster (for example is't still (re)provisioning and the status is not set yet).

Reply:
Done in 91b78d30c6d4c8f81200275cffe6abcef680eb9e